docs/decisions/2023-06-05_github_merge_queue.md

```diff
@@ -0,0 +1,49 @@
+# 2023-06-05 GitHub Merge Queue
+
+* status: proposed
+* date: 2023-06-05
+* authors: Brian Samek
+
+## Context and Problem Statement
+
+* [Epic](https://go/github-merge-queue)
+* [Scope](https://go/github-merge-queue-scope)
+* [Design](https://go/github-merge-queue-design)
+
+GitHub merge queue integrates natively with GitHub Actions. Integration with
+other CI systems exists but is lightly documented. The
+[docs](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue#triggering-merge-group-checks-with-other-ci-providers)
+state that a CI system should listen for branches matching a pattern and that
+the PR will be merged once required checks pass.
+
+The purpose of this ADR is to keep a record of the details of the integration
+between the GitHub merge queue and Evergreen for later reference.
+
+## Architectural Overview
+
+Evergreen listens for new GitHub merge queue items, creates and runs versions
+for these items, and posts the results of these versions to GitHub.
+
+### Listen
+
+Evergreen listens for new branches matching `gh-readonly-queue/{base_branch}`.
+
+### Create version
+
+Evergreen creates a new version with a new requester type "github-merge-queue".
```

Comment:
Will the list of checks included in these versions still be determined by a project setting, the way it is now?

Reply:
Yes. Added.